pkg/announce/github_page.go

```diff
@@ -291,19 +292,37 @@ func UpdateGitHubPage(opts *GitHubPageOptions) (err error) {
 		return errors.New("tag not found while trying to publish release page")
 	}
 
-	// Get the release we are looking for
-	releases, err := gh.Releases(opts.Owner, opts.Repo, true)
+	// Get the release we are looking for. We only need to fetch prereleases
+	// if the release is a prerelease.
+	releases, err := gh.Releases(opts.Owner, opts.Repo, isPrerelease)
 	if err != nil {
 		return fmt.Errorf("listing the repositories releases: %w", err)
 	}
 
-	// Does the release exist yet?
+	// Does the release exist yet and should it be marked as latest?
 	var releaseID int64
 	commitish := ""
+	// No pre-release should ever be marked as "latest"
+	markAsLatest := !isPrerelease
+
 	for _, release := range releases {
 		if release.GetTagName() == opts.Tag {
 			releaseID = release.GetID()
 			commitish = release.GetTargetCommitish()
+		} else if markAsLatest {
+			// If this release is not identical to the one being cut right now,
+			// we will check if *our* release is lower than the release from the loop.
+			// If the first page of releases does not include a release that is
+			// greater than *our* release, we can assume that ours will be the
+			// latest release right now.
+			releaseSemver, err := util.TagStringToSemver(release.GetTagName())
+			if err != nil {
+				return fmt.Errorf("parsing existing release tags as semver: %w", err)
+			}
+
+			if semver.LE(releaseSemver) {
+				markAsLatest = false
+			}
```

Comment:
I think the current implementation is fine because we already have all the releases at hand. Doing another call to `https://dl.k8s.io/release/stable.txt` seems to be not required for my taste especially since we update the github release here.

Reply:
I was worried about pagination making a problem here, but looks like this API call is not paginated, so we don't have to worry about this. Let's go with the current implementation.